test/node/error_test.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const BSON = require('../register-bson');
+const BSONTypeError = BSON.BSONTypeError;
+const BSONError = BSON.BSONError;
+
+describe('BSONTypeError', function () {
+  it('should evaluate true on instanceof BSONTypeError and TypeError', function () {
+    const bsonTypeErr = new BSONTypeError();
+    expect(bsonTypeErr instanceof BSONTypeError).to.be.true;
```

Comment:
seems to be missing `instanceof TypeError` and the case below is missing `instanceof Error` check